src/common/btree.c

```diff
@@ -48,6 +48,8 @@ enum btr_probe_rc {
 	PROBE_RC_NONE,
 	PROBE_RC_OK,
 	PROBE_RC_ERR,
+	PROBE_RC_INPROGRESS,
+	PROBE_RC_UNAVAILABLE	= PROBE_RC_UNKNOWN,
```

Comment:
For the availability check, it doesn't use PROBE_RC_UNKNOWN, then define PROBE_RC_UNAVAILABLE as PROBE_RC_UNKNOWN is enough for availability check.

Reply:
About the btree/evtree record availability check, the current approach is more focus on the efficiency. As I mentioned before, it is not impossible to re-implement that similar as you described, but that will introduce more inefficiency and control points. For example, for an iteration, the caller may have to check the availability for each returned position by itself, and if unavailable, then re-call the probe() API, that is inefficient than making the probe API to direct move its cursor inside the API as current implementation does. On the other hand, there may be multiple probe()/insert()/delete() API calls outside of the common btree/evtree logic, then each of them needs to check the availability, that make the code looks more redundant and unclean to me. In fact, we already have 'visibility' concept for the btree/evtree in current master implementation, the DTX related 'availability' can be regarded as another dimension of 'visibility'. From such view, I think current implementation is also reasonable.

As for the 'ALB_AVAILABLE_DIRYT/CLEAN', it is mainly exported for 'garbage collection' purpose, or say used by the new aggregation logic to make the caller to know whether the target record is garbage or not. If without such requirement, we can directly name it as "ALB_AVAILABLE".

Reply:
I think both RC_INPROGRESS and RC_UNAVAIL can be just an error code and only internally checked right? we probably should just use error code.

The only reason I originally defined RC_ERR is because there is no actual error code can be returned, but now it has different error code so we should just return error code.
But this can be cleaned in the future